### PR TITLE
Add the `cols_merge_n_pct()` function

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR adds the `cols_merge_n_pct()` function, which merges two columns together: a count column and a percentage column. This function will autohide the `col_pct` column by default (but that can be changed by setting `autohide = FALSE`). Different combinations of `NA` are given special consideration much like `col_vals_uncert()` function. Any zero values in the `col_n` column will be formatted solely as `0`.

Fixes: #651

### CODE REVIEW

File: R/modify_columns.R [hunk 1]
```diff
@@ -1036,6 +1036,97 @@ cols_merge_resolver <- function(data, col_begin, col_end, sep) {
...
+#'   component.
+#' @param autohide An option to automatically hide the column specified as
+#'   `col_pct`. Any columns with their state changed to hidden will behave
+#'   the same as before, they just won't be displayed in the finalized table.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @family Modify Columns
+#' @section Function ID:
+#' 4-10
```
Comment (jcheng5) [inline R/modify_columns.R:1091]: Example

File: tests/testthat/test-cols_merge.R [hunk 2]
```diff
@@ -434,3 +434,60 @@ test_that("the `cols_merge_range()` function works correctly", {
...
+
+  # Check that specific suggested packages are available
+  check_suggests()
+
+  tbl_n_pct <-
+    dplyr::tribble(
+      ~a, ~b,
+      1,  0.0714,
+      5,  0.3571,
+      0,  0.0,
```
Comment (jcheng5) [inline tests/testthat/test-cols_merge.R:449]: Let's add a line for `0, NA` and `NA, 0`

File: R/modify_columns.R (hunk 1 above)
Comment (rich-iannone) [reply R/modify_columns.R:1136]: Example now added.

File: tests/testthat/test-cols_merge.R (hunk 2 above)
Comment (rich-iannone) [reply tests/testthat/test-cols_merge.R:449]: Now added.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**